src/Provider/ActiveMQ.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace Riverline\WorkerBundle\Provider;
+
+use Riverline\WorkerBundle\Queue\Queue;
+use Stomp\Client;
+use Stomp\Network\Connection;
+use Stomp\StatefulStomp;
+use Stomp\Transport\Message;
+
+/**
+ * Class ActiveMQ
+ * @package Riverline\WorkerBundle\Provider
+ */
+class ActiveMQ extends AbstractBaseProvider
+{
+    /**
+     * @var StatefulStomp
+     */
+    private $stomp;
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * ActiveMQ constructor.
+     *
+     * @param string $brokerUri
+     * @param string $login
+     * @param string $password
+     */
+    public function __construct($brokerUri, $login, $password)
+    {
+        $connection = new Connection($brokerUri);
+        $connection->setReadTimeout(5, 0);
+
+        $this->client = new Client($connection);
+        $this->client->setLogin($login, $password);
+        $this->client->setSync(true);
+
+        $this->stomp = new StatefulStomp($this->client);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count($queueName)
+    {
+        $tempQueueName = sprintf("/temp-queue/%s", uniqid());
+
+        $this->client->sendFrame(new Message(
+            "",
+            [
+                "destination" => $tempQueueName,
+            ]
+        ), true);
+
+        $this->client->sendFrame(new Message(
+            "",
+            [
+                "reply-to"    => $tempQueueName,
+                "destination" => "ActiveMQ.Statistics.Destination.".$queueName,
+            ]
+        ), true);
+
+        $subscriptionId = $this->stomp->subscribe($tempQueueName);
+
+        // Ignore first frame
+        $this->stomp->read();
+        $frame = $this->stomp->read();
+        $this->stomp->unsubscribe($subscriptionId);
+
+        if (false === $frame) {
+            // No queue existing
+            return 0;
+        }
+
+        $statistics = new \SimpleXMLElement($frame->body);
+        if (false === $statistics) {
+            throw new \RuntimeException("Unable to decode statistics frame body");
+        }
+
+        $elements = $statistics->xpath("/map/entry[string='size'][1]");
+        $element  = array_shift($elements);
+
+        return intval($element->long);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createQueue($queueName, array $queueOptions = [])
+    {
+        return new Queue($queueName, $this);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteQueue($queueName)
+    {
+        // Queue is deleted automatically when no more messages exists in queue
+        while ($this->get($queueName, 1) !== null) {
```

Comment:
May be dangerous if the user have 1M messages pending.
I prefer that you throw an exception because it is not supported by the protocol.